rabbit-openqa.py

```diff
@@ -0,0 +1,209 @@
+                                no_ack=True)
+
+        print(' [*] Waiting for logs. To exit press CTRL+C')
+
+    def add(self, project):
+        self.projects.append(project)
+
+    def on_published_repo(self, payload):
+        for p in self.projects:
+            p.check_published_repo(str(payload['project']), str(payload['repo']), str(payload['buildid']))
+
+    def on_openqa_create(self, payload):
+        for p in self.projects:
+            p.openqa_create(str(payload.get('ISO', '')), str(payload['TEST']), str(payload['MACHINE']), str(payload['id']))
+
+    def on_openqa_restart(self, payload):
+        print(payload)
+
+    def on_openqa_done(self, payload):
+        for p in self.projects:
+            p.openqa_done(str(payload.get('ISO', '')), payload['TEST'], payload['MACHINE'], payload['id'], payload['result'])
+
+    def on_message(self, unused_channel, method, properties, body):
+        if method.routing_key == '{}.obs.repo.published'.format(amqp_prefix):
+            self.on_published_repo(json.loads(body))
+
+        if method.routing_key == '{}.openqa.job.done'.format(amqp_prefix):
+            self.on_openqa_done(json.loads(body))
+        if method.routing_key == '{}.openqa.job.create'.format(amqp_prefix):
+            self.on_openqa_create(json.loads(body))
+        if method.routing_key == '{}.openqa.job.restart'.format(amqp_prefix):
+            self.on_openqa_restart(json.loads(body))
+
+    def listen(self):
+        self.channel.start_consuming()
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description='Bot to sync openQA status to OBS')
+    parser.add_argument("--apiurl", '-A', type=str, default='https://api.opensuse.org', help='API URL of OBS')
+    parser.add_argument('-s', '--staging', type=str, default=None,
+                        help='staging project letter')
+    parser.add_argument('-f', '--force', action='store_true', default=False,
+                        help='force the write of the comment')
+    parser.add_argument('-p', '--project', type=str, default='Factory',
+                        help='openSUSE version to make the check (Factory, 13.2)')
+    parser.add_argument('-d', '--debug', action='store_true', default=False,
+                        help='enable debug information')
+
+    args = parser.parse_args()
+
+    osc.conf.get_config()
+    osc.conf.config['debug'] = args.debug
+
+    apiurl = args.apiurl
+
+    if apiurl.endswith('suse.de'):
+        amqp_prefix = 'suse'
+        amqp_url = "amqps://suse:suse@rabbit.suse.de?heartbeat_interval=15"
+    else:
+        amqp_prefix = 'opensuse'
+        amqp_url = "amqps://opensuse:opensuse@rabbit.opensuse.org?heartbeat_interval=15"
+
+    l = Listener(amqp_prefix, amqp_url)
+    if amqp_prefix == 'opensuse':
+
+        class Leap15(Project):
+            def __init__(self, name):
+                super(name)
+
+        l.add(Leap15('openSUSE:Leap:15.1'))
+    else:
+        class Sle15(Project):
```

Comment:
Would seem this would be fit better in remote config rather than essentially hard-coding.

Reply:
Yes, but I'm super afraid of writing one regexp to rule it all. I guess it's possible, but having a function was easier in the end.

We would still need to define somewhere what projects to look at.

Reply:
I think I found a way: 111aabb

Question remains how to avoid hardcoding project lists. I guess through an attribute. And if I need an attribute anyway, I can store the replacement pattern in it right away